src/regression.js

```diff
@@ -136,6 +150,8 @@
       var gradient;
       var intercept;
       var len = data.length;
+      var nonNullLen = 0; // the number of non-null points
```

Comment:
Ok, a couple things here:

1. It's always good to be smart about application performance, but even more important is to convey a clear message to the next developer.
2. To me, it's far clearer to state "only retain the non null data", then "operate only on the non null data" than it is to add a second counter.
3. Given that, and given the fact that there are many, many loops in here which don't  seem to cause performance degradations, I would ask myself to prove that this optimization is worth the cost in lack of clarity. If it is, great, then I reinforce this optimization with tests.